### PR TITLE
Add option to set search paths in preferences.py

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

File: operators/__init__.py [hunk 1]
```diff
@@ -118,12 +118,55 @@ def execute(self, context):
...
-    ASSSS_OT_add_sss_set,
-    ASSSS_OT_remove_sss_set
-)
+class ASSPREF_OT_add_searchpath(bpy.types.Operator):
+    """
+    Operator for adding searchpaths
+    """
+
+    bl_label = "Add Set"
+    bl_description = "Add new searchpath"
```
Comment (dictoon) [inline operators/__init__.py]: *search path* in user-visible strings.

File: operators/__init__.py [hunk 2]
```diff
@@ -118,12 +118,55 @@ def execute(self, context):
...
+    bl_idname = "appleseed.add_searchpath"
+
+    def execute(self, context):
+        collection = context.preferences.addons['blenderseed'].preferences.search_paths
+
+        collection.add()
+        num = len(collection)
+        collection[num - 1].name = f"Searchpath {num}"
```
Comment (dictoon) [inline operators/__init__.py]: If this string is user-visible, it should be *Search Path {num}* or *Search path {num}* (depending on whether we use title case or not).

File: operators/__init__.py [hunk 3]
```diff
@@ -118,12 +118,55 @@ def execute(self, context):
...
+        return {'FINISHED'}
+
+
+class ASSPREF_OT_remove_searchpath(bpy.types.Operator):
+    """
+    Operator for removing searchpaths
+    """
+
+    bl_label = "Remove Set"
+    bl_description = "Remove searchpath"
```
Comment (dictoon) [inline operators/__init__.py]: *search path*

File: operators/__init__.py [hunk 4]
```diff
@@ -118,12 +118,55 @@ def execute(self, context):
...
 
-classes = (
-    ASPP_OT_add_postproc_stage,
-    ASPP_OT_remove_postproc_stage,
-    ASSSS_OT_add_sss_set,
-    ASSSS_OT_remove_sss_set
-)
+class ASSPREF_OT_add_searchpath(bpy.types.Operator):
+    """
+    Operator for adding searchpaths
```
Comment (dictoon) [inline operators/__init__.py]: *search paths*

File: operators/__init__.py [hunk 5]
```diff
@@ -118,12 +118,55 @@ def execute(self, context):
...
+    """
+
+    bl_label = "Add Set"
```
Comment (dictoon) [inline operators/__init__.py]: That seems like a copy/paste error?

File: operators/__init__.py [hunk 6]
```diff
@@ -118,12 +118,55 @@ def execute(self, context):
...
+
+        return {'FINISHED'}
+
+
+class ASSPREF_OT_remove_searchpath(bpy.types.Operator):
+    """
+    Operator for removing searchpaths
+    """
+
+    bl_label = "Remove Set"
```
Comment (dictoon) [inline operators/__init__.py]: Same here.

File: operators/__init__.py [hunk 7]
```diff
@@ -118,33 +118,33 @@ def execute(self, context):
         return {'FINISHED'}
 
 
-class ASSPREF_OT_add_searchpath(bpy.types.Operator):
+class ASSPREF_OT_add_search_path(bpy.types.Operator):
     """
     Operator for adding searchpaths
```
Comment (est77) [inline operators/__init__.py]: Change the doc string too

File: operators/__init__.py [hunk 8]
```diff
@@ -118,33 +118,33 @@ def execute(self, context):
...
-        collection[num - 1].name = f"Searchpath {num}"
+        collection[num - 1].name = f"{num}"
 
         return {'FINISHED'}
 
 
-class ASSPREF_OT_remove_searchpath(bpy.types.Operator):
+class ASSPREF_OT_remove_search_path(bpy.types.Operator):
     """
     Operator for removing searchpaths
```
Comment (est77) [inline operators/__init__.py]: Here too

File: preferences.py [hunk 9]
```diff
@@ -128,13 +86,19 @@ def draw(self, context):
...
 
         if self.search_paths:
             current_set = self.search_paths[self.path_index]
-            layout.prop(current_set, "file_path", text="Filepath")
+            layout.prop(current_set, "name", text="Filepath")
+
+        global path_added
 
+        if path_added is True:
+            layout.label(text="Restart Blender to load new search path!", icon="ERROR")
```
Comment (est77) [inline preferences.py]: Maybe: "Changes to search paths will only take effect after restarting Blender" ?

**Pull request merged**